llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -10659,8 +10699,8 @@ bool LoopVectorizePass::processLoop(Loop *L) {
     bool ForceVectorization =
         Hints.getForce() == LoopVectorizeHints::FK_Enabled;
     if (!ForceVectorization &&
-        !areRuntimeChecksProfitable(Checks, VF, L, PSE, SEL,
-                                    CM.getVScaleForTuning())) {
+        !isOutsideLoopWorkProfitable(Checks, VF, CM, PSE,
```

Comment:
passing CM is now unnecessary/unrelated?

Reply:
Oh I read this after reading your 'unrelated' comments above, which I found confusing at first. The problem is that so many of the objects required to calculate the early exit cost live in the cost model class so it seemed simpler to pass that in here. Perhaps I'm missing something, but it wasn't clear from your comments above what you wanted me to do here? I thought originally you were suggesting I move the creation of the `VPCostContext` object into `isOutsideLoopWorkProfitable`, but are you suggesting I create the `VPCostContext` class here instead and pass it to `isOutsideLoopWorkProfitable`?

Reply:
Adding here was my thinking, to avoid spreading further references of the legacy CM.